compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2772,6 +2772,44 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         Some(r)
     }
 
+    /// make sure that we are in the condition to suggest the blanket implementation, if we are in the
+    /// case of suggest it, use the function `err_reporter` to report the error or suggestion.
+    // FIXME: The copy of TyCtxt it is a bad think in this case?
+    fn maybe_lint_blanket_trait_impl(
+        &self,
+        self_ty: &hir::Ty<'_>,
+        err_reporter: fn(TyCtxt<'tcx>, &hir::Ty<'_>, &str, &str, Vec<(Span, String)>),
+    ) {
+        let tcx = self.tcx();
+        let parent_id = tcx.hir().get_parent_item(self_ty.hir_id);
+        let child_ty = self_ty;
+        if let hir::Node::Item(hir::Item {
+            kind: hir::ItemKind::Impl(hir::Impl { self_ty, of_trait: Some(trait_ref), .. }),
+            ..
+        }) = tcx.hir().get_by_def_id(parent_id)
+        {
+            let hir::TraitRef { path: hir::Path { span, .. }, .. } = trait_ref;
+            let is_local =
+                if let Some(def_id) = trait_ref.trait_def_id() { def_id.is_local() } else { false };
+            if is_local {
+                let blanket_msg = "for local traits use the blanket implementation";
```

Comment:
This message should be more of a suggestion, for example "use a blanket implementation to implement {trait} for all types that also implement {other_trait}".

Reply:
This is a great error message, thanks for the suggestion!